src/test/java/htsjdk/samtools/cram/build/ContainerFactoryTest.java

```diff
@@ -1,148 +1,297 @@
+        final List<Integer> expectedContainerRecordCounts,
+        final List<ReferenceContext> expectedContainerReferenceContexts
+    ) {
+        final CRAMEncodingStrategy cramEncodingStrategy =
+                new CRAMEncodingStrategy()
+                    .setMinimumSingleReferenceSliceSize(readsPerSlice)
+                    .setReadsPerSlice(readsPerSlice)
+                    .setSlicesPerContainer(slicesPerContainer);
+        final ContainerFactory containerFactory = new ContainerFactory(
+                CRAMStructureTestHelper.SAM_FILE_HEADER,
+                cramEncodingStrategy,
+                CRAMStructureTestHelper.REFERENCE_SOURCE);
+        final List<Container> containers = CRAMStructureTestHelper.createContainers(containerFactory, samRecords);
 
-        final ReferenceContext refContext = new ReferenceContext(0);
-        final int alignmentStart = 1;
-        final int alignmentSpan = READ_LENGTH_FOR_TEST_RECORDS;
-        int recordCount = 1;
-        int globalRecordCount = 0; // first container - no records yet
-        CRAMStructureTestUtil.assertContainerState(containers.get(0), refContext, alignmentStart, alignmentSpan,
-                recordCount, READ_LENGTH_FOR_TEST_RECORDS * recordCount, globalRecordCount,
-                firstContainerByteOffset);
+        Assert.assertEquals(containers.size(), expectedContainerCount);
+        Assert.assertEquals(containers.size(), expectedContainerReferenceContexts.size());
 
-        // when other refs are added, subsequent containers are multiref
-
-        recordCount++;  // this container has 2 records
-        globalRecordCount = containers.get(0).nofRecords;   // we've seen 1 record before this container
-        CRAMStructureTestUtil.assertContainerState(containers.get(1), ReferenceContext.MULTIPLE_REFERENCE_CONTEXT,
-                Slice.NO_ALIGNMENT_START, Slice.NO_ALIGNMENT_SPAN, recordCount,
-                READ_LENGTH_FOR_TEST_RECORDS * recordCount, globalRecordCount, firstContainerByteOffset + 1);
-
-        recordCount++;  // this container has 3 records
-        globalRecordCount = containers.get(0).nofRecords + containers.get(1).nofRecords;    // we've seen 3 records before this container
-        CRAMStructureTestUtil.assertContainerState(containers.get(2), ReferenceContext.MULTIPLE_REFERENCE_CONTEXT,
-                Slice.NO_ALIGNMENT_START, Slice.NO_ALIGNMENT_SPAN, recordCount,
-                READ_LENGTH_FOR_TEST_RECORDS * recordCount, globalRecordCount, firstContainerByteOffset + 2);
+        for (int i = 0; i < containers.size(); i++) {
+            final Container container = containers.get(i);
+            Assert.assertEquals(
+                    (Integer) container.getContainerHeader().getNumberOfRecords(),
+                    expectedContainerRecordCounts.get(i));
+            Assert.assertEquals(container.getAlignmentContext().getReferenceContext(), expectedContainerReferenceContexts.get(i));
+        }
     }
+
 }
```

Comment:
I didn't see any tests with name-sorted reads -- can you add some?

Reply:
Yeah, that would be a good addition.

Reply:
Done.